pre_commit_all_the_way_down.py

```diff
@@ -0,0 +1,126 @@
+import argparse
+import re
+import subprocess
+import sys
+from pathlib import Path
+from re import Match
+from tempfile import TemporaryDirectory
+from textwrap import dedent, indent
+from typing import Optional, Sequence
+
+BLOCK_TYPES = "(code|code-block|sourcecode|ipython)"
+PY_LANGS = "(python|py|sage|python3|py3|numpy)"
+DOCTEST_TYPES = "(testsetup|testcleanup|testcode)"
+RST_RE = re.compile(
+    rf"(?P<before>"
+    rf"^(?P<indent> *)\.\. ("
+    rf"jupyter-execute::|"
+    rf"python-script::|"
+    rf"{BLOCK_TYPES}:: {PY_LANGS}|"
+    rf"{DOCTEST_TYPES}::.*"
+    rf")\n"
+    rf"((?P=indent) +:.*\n)*"
+    rf"\n*"
+    rf")"
+    rf"(?P<code>(^((?P=indent) +.*)?\n)+)",
+    re.MULTILINE,
+)
+INDENT_RE = re.compile("^ +(?=[^ ])", re.MULTILINE)
+TRAILING_NL_RE = re.compile(r"\n+\Z", re.MULTILINE)
+
+
+def fix_block(block: str, whitelist: Sequence[Optional[str]]):
+    """
+    Fix a code block.
+
+    Parameters
+    ----------
+    block : str
+        The block to fix
+
+    Notes
+    -----
+    The code block should have no leading indentation and be valid
+    Python.
+    """
+
+    def _pre_commit_helper(fname: str, hook_id: Optional[str]):
+        args = ["pre-commit", "run"]
+        if hook_id is not None:
+            args.append(hook_id)
+        args.extend(("--files", fname))
+        try:
+            subprocess.run(args, check=True, capture_output=True)
```

Comment:
I see in the other PR that it looks like you want to support python 3.6, though it's not clear to me why (this program will be installed in isolation via pre-commit any way so we're free to immediately use the latest version), know that this line isn't compatible with 3.6: the `capture_output` option was added in 3.7 (see https://docs.python.org/3/library/subprocess.html)

Reply:
Good catch! I'll update the other as I don't really want support for 3.6.